### PR TITLE
Change `NU_LIB_DIRS` not to depend on `$nu.config-path`

### LINKED ISSUE #8870: NU_LIB_DIRS environment variable no longer working
### Describe the bug

Setting `let-env NU_LIB_DIRS` inside `env.nu` has no effect. Only `const NU_LIB_DIRS` works.

### How to reproduce

1. Inside `env.nu`: Set `let-env NU_LIB_DIRS = [  ($nu.config-path | path dirname | path join 'scripts') ]`
2. `const NU_LIB_DIRS` is not set anywhere
3. Put a file (`script.nu`) to the scripts folder
4. `source script.nu` inside `config.nu`
5. Error: File not found

### Expected behavior

Should work the same as before.

### Screenshots

_No response_

### Configuration

| key                | value                                                                                       |
| ------------------ | ------------------------------------------------------------------------------------------- |
| version            | 0.78.1                                                                                      |
| branch             | main                                                                                        |
| commit_hash        | 99c0a2575f22c754cc4e2ee14aa7e539dcc364b9                                                    |
| build_os           | linux-x86_64                                                                                |
| build_target       | x86_64-unknown-linux-gnu                                                                    |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)                                                         |
| rust_channel       | 1.66.1-x86_64-unknown-linux-gnu                                                             |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)                                                         |
| build_time         | 2023-04-13 00:14:22 +03:00                                                                  |
| build_rust_channel | release                                                                                     |
| features           | default, zip                                                                                |
| installed_plugins  | from eml, from ics, from ini, from vcf, gstat, inc, query, query json, query web, query xml |

### Additional context

### PR DESCRIPTION
**Pull request by ito-hiroki**
# Description

fixed #8870
In `default_env.nu`, environment variable `NU_LIB_DIRS` is derived from `$nu.config-path` now. If the config file is changed (e.g., by using `--column`), `NU_LIB_DIRS` is also changed.
This PR adds `default-config-dir` to $nu and fix `default_env.nu` to use `default-config-dir` for `NU_LIB_DIRS` definition.

# User-Facing Changes

Users can use `$nu.default-config-dir` to specify the default config directory.

# Tests + Formatting

```bash
$ use toolkit.nu  # or use an `env_change` hook to activate it automatically
$ toolkit check pr

- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :green_circle: `toolkit test`
- :green_circle: `toolkit test stdlib`
```

# After Submitting

nothing

### CODE REVIEW

**Comment by fdncred** [comment]
I'm fine with following @kubouch's lead here but this seems weird to me, even after following the other conversations.

1. If someone is specifying a `--config /some/alternate/config.nu` it seems reasonable to think that all their config settings, including NU_LIB_DIRS, would be based off of that new path, instead of "hard-coded" to the default path.
2. Having said that, it's easy enough to change the env.nu to point to `$nu.config-path` instead of `$nu.default-config-path`

File: crates/nu-engine/src/nu_variable.rs [hunk 1]
```diff
@@ -25,7 +25,13 @@ pub struct NuVariable {
 
 impl LazyRecord for NuVariable {
     fn column_names(&self) -> Vec<&'static str> {
-        let mut cols = vec!["config-path", "env-path", "history-path", "loginshell-path"];
+        let mut cols = vec![
+            "default-config-dir",
```
Comment (fdncred) [inline crates/nu-engine/src/nu_variable.rs:29]: can we change this to `default-config-path` so we stay consisten with `-path`?

Comment (kubouch) [reply crates/nu-engine/src/nu_variable.rs:29]: `config-path` points to the config file while the new path points at a directory. That's why originally suggested -dir, not -path name.

Comment (fdncred) [reply crates/nu-engine/src/nu_variable.rs:29]: ok, it just looked out of place where everything else is path, but i see the distinction now. thx.

**Comment by ito-hiroki** [comment]
> If someone is specifying a --config /some/alternate/config.nu it seems reasonable to think that all their config settings, including NU_LIB_DIRS, would be based off of that new path, instead of "hard-coded" to the default path.

I agree with you.
Following your comment, I think that only adding `default-config-path` (not to change `default_env.nu`) is enough in this PR.
Should we change `NU_LIB_DIRS` in `default_env.nu`?

**Comment by sholderbach** [comment]
Is this a breaking change?

**Comment by ito-hiroki** [comment]
Users using a custom config directory will be affected when they replace their `env.nu` to `default_env.nu`. One option is not to change `default_env.nu` to respect the existing configuration, but which is the best?

**Comment by kubouch** [comment]
I'd mark it as a breaking change because it changes default env.nu. But actually, what this PR does is that it brings the defalt behavior back after https://github.com/nushell/nushell/pull/8792 which caused the breaking change.

**Comment by kubouch** [comment]
@fdncred Are you OK with this PR or do you still have some concerns?

**Comment by fdncred** [comment]
@kubouch I'm good if you're good.

**Comment by sholderbach** [comment]
> I'd mark it as a breaking change because it changes default env.nu. But actually, what this PR does is that it brings the defalt behavior back after #8792 which caused the breaking change.

I don't see how #8792 should have changed the `NU_LIB_DIRS` behavior. It only affects the behavior of `config nu` and `config env`?

Might this be related to one of the IDE PRs?

**Comment by kubouch** [comment]
@sholderbach [This line](https://github.com/nushell/nushell/blob/4ecec5922473dd8e9e07cefd1e2ff258546a91d3/crates/nu-utils/src/sample_config/default_env.nu#L75) used to depend on `$nu.config-path` which became volatile after the PR.

**Pull request merged**